### PR TITLE
[assertion] create parse_public_key_credential for AccountAuthenticator

### PR DESCRIPTION
**Pull request by hariria**
# Description

This contains all of the standalone webauthn assertion
functionality for usage with Aptos' `AccountAuthenticator`
for webauthn transactions.

Mostly it uses and refactors `verify_assertion` from `core.rs`
to statelessly support assertion verification, without
the need for a `WebauthnCore` struct. We exclude many of
the verification steps from `verify_credential_internal`
as they are not needed.

Ultimately `parse_public_key_credential` returns a tuple, containing the verification_data and raw, fixed length signature which will be used in the `AccountAuthenticator` to verify the signature with the public key.

Currently `P256` is the only signature scheme supported, but more could be added in the future.

# Test Plan
- [x] Includes tests for both `parse_public_key_credential` and `p256_der_to_raw_signature`
- [x] cargo test, clippy, fmt has been run and passes

### CODE REVIEW

File: webauthn-rs-core/src/assertion.rs [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+    // Verify that the value of C.type is the string webauthn.get.
+    if c.type_ != "webauthn.get" {
+        return Err(WebauthnError::InvalidClientDataType);
+    }
+
+    // Let hash be the result of computing a hash over the cData using SHA-256.
+    let client_data_json_hash = compute_sha256(data.client_data_bytes.as_slice());
+
+    // Using the credential public key, verify that sig is a valid signature
```
Comment (mstraka100) [inline webauthn-rs-core/src/assertion.rs]: Change this comment so it doesn't imply you're validating the signature here

File: webauthn-rs-core/src/assertion.rs [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+        assert!(fixed_len_sig.is_ok());
+        assert_eq!(fixed_len_sig.unwrap().to_bytes().to_vec().len(), 64);
+    }
+
+    #[test]
+    fn test_authentication() {
+        let _ = tracing_subscriber::fmt::try_init();
+
+        // Create the fake credential that we know is associated
```
Comment (mstraka100) [inline webauthn-rs-core/src/assertion.rs]: What does "fake" here mean?

File: webauthn-rs-core/src/assertion.rs [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+        let der_sig = data.signature;
+        let fixed_len_sig = p256_der_to_raw_signature(&der_sig.as_slice());
+
+        assert!(fixed_len_sig.is_ok());
+        assert_eq!(fixed_len_sig.unwrap().to_bytes().to_vec().len(), 64);
+    }
+
+    #[test]
+    fn test_authentication() {
```
Comment (mstraka100) [inline webauthn-rs-core/src/assertion.rs]: Might be good to have a similar test that you know will fail (to make sure it doesn't succeed when it shouldn't)

**Review by mstraka100 — APPROVED** [review]
Left a few comments. Looks good overall

File: webauthn-rs-core/src/assertion.rs (hunk 2 above)
Comment (hariria) [reply webauthn-rs-core/src/assertion.rs]: I can change it to say `test credential` instead

File: webauthn-rs-core/src/assertion.rs (hunk 3 above)
Comment (hariria) [reply webauthn-rs-core/src/assertion.rs]: Good idea

File: webauthn-rs-core/src/assertion.rs (hunk 2 above)
Comment (mstraka100) [reply webauthn-rs-core/src/assertion.rs]: I think that's more clear

**Pull request merged**